src/v/kafka/server/handlers/details/security.h

```diff
@@ -150,4 +151,155 @@ inline security::acl_binding to_acl_binding(const creatable_acl& acl) {
     return security::acl_binding(std::move(pattern), std::move(entry));
 }
 
+/*
+ * build resource pattern filter bits
+ */
+inline security::resource_pattern_filter
```

Comment:
there are a ton of these cool funcs, should we have tests for these? 

this in particular has more complex casing/matching.

Reply:
yeh, that's a good idea

Reply:
fixed